### PR TITLE
Added simple summary output

### PR DESCRIPTION
**Pull request by k-dominik**
The build process takes time, and instead of having to scroll through a lot of text, I thought it would be nice to have short summary.

Report file looks like the following

```
built:
- {pakage_name: ilastik-meta, recipe_build_string: 0_g1355659, recipe_version: 1.3.1b2}
- {pakage_name: ilastik-launch, recipe_build_string: '7', recipe_version: '0.1'}
- {pakage_name: ilastik-dependencies-no-solvers, recipe_build_string: '0', recipe_version: 1.3.1}
- {pakage_name: ilastik-dependencies, recipe_build_string: '0', recipe_version: 1.3.1}
found:
- {pakage_name: lemon, recipe_build_string: '5', recipe_version: 1.2.4}
- {pakage_name: vigra, recipe_build_string: np114py36_8_gfb42744, recipe_version: 1.11.1.post8}
- {pakage_name: dpct, recipe_build_string: py36_2_gc6745f0, recipe_version: 1.2.post18}
- {pakage_name: yapsy, recipe_build_string: py36_0, recipe_version: 1.10.423}
- {pakage_name: marching_cubes, recipe_build_string: py36_2_g372a1b5, recipe_version: 0.0.post37}
- {pakage_name: hytra, recipe_build_string: py36_0_g52df98b, recipe_version: 1.1.4.post21}
- {pakage_name: ilastik-feature-selection, recipe_build_string: py36_0_g413714a, recipe_version: 0.123.post13}
- {pakage_name: ilastiktools, recipe_build_string: np114py36_2_gae9e090, recipe_version: 0.2.post20}
- {pakage_name: ilastikrag, recipe_build_string: py36_0_g894ba23, recipe_version: 0.1.3.post8}
- {pakage_name: mamutexport, recipe_build_string: py36_0_g5023d4d, recipe_version: 0.2.1.post5}
- {pakage_name: tifffile, recipe_build_string: np114py36_2_g7c3bb9e, recipe_version: 0.4.post2}
- {pakage_name: pytiff, recipe_build_string: np114py36_1_g60a58ae, recipe_version: 0.7.2}
- {pakage_name: numpy-allocation-tracking, recipe_build_string: py36np114_0_g6300784,
  recipe_version: 0.2.post4}
- {pakage_name: wsdt, recipe_build_string: py36_0_g76f938f, recipe_version: 0.2.post16}
- {pakage_name: fastfilters, recipe_build_string: py36_1_gce61889, recipe_version: 0.2.6.post13}
- {pakage_name: nifty, recipe_build_string: py36_2_g13cedaf, recipe_version: 0.18.1.post21}
- {pakage_name: opengm-structured-learning-headers, recipe_build_string: 5_ga0797cd, recipe_version: 2.3.4}
- {pakage_name: cplex-shared, recipe_build_string: '9', recipe_version: '0.1'}
- {pakage_name: gurobi-symlink, recipe_build_string: '3', recipe_version: '0.1'}
- {pakage_name: multi-hypotheses-tracking-with-gurobi, recipe_build_string: py36_2_gb63485b,
  recipe_version: 1.1.3.post34}
- {pakage_name: multi-hypotheses-tracking-with-cplex, recipe_build_string: py36_2_gb63485b,
  recipe_version: 1.1.3.post34}
- {pakage_name: nifty-with-gurobi, recipe_build_string: py36_2_g13cedaf, recipe_version: 0.18.1.post21}
- {pakage_name: nifty-with-cplex, recipe_build_string: py36_2_g13cedaf, recipe_version: 0.18.1.post21}
recipe_specs_path: ilastik-recipe-specs.yaml
selected_recipies: []
duration: '0:16:49.249799'
end_time: '2018-05-30T15:50:02'
start_time: '2018-05-30T15:33:13'
```

### CODE REVIEW

File: build-recipes.py [hunk 1]
```diff
@@ -110,11 +111,31 @@ def main():
...
+
+    end_time = datetime.datetime.now()
+    result['end_time'] = end_time.isoformat(timespec='seconds')
+    result['duration'] = str(end_time - start_time)
+    with open(result_file, 'w') as f:
+        yaml.dump(result, f, default_flow_style=False)
 
     print("--------")
-    print("DONE")
+    print(f"DONE, Result written to {result_file}")
```
Comment (FynnBe) [inline build-recipes.py]: Maybe also write the summary to the console?

**Comment by FynnBe** [comment]
Oh.. I have another idea, we might add in: Check if such a summary file exists and then skip those packages (in a group 'found_locally' or something). 
This would avoid the hastle of uncommenting everything one has already built to avoid the long wait for the conda render command to finish.

**Comment by k-dominik** [comment]
Didn't you also have the idea of a `--skip-until` option. @FynnBe ? When it comes to not check certain builds I would prefer that - I think it's harder to get something wrong with this than with the possibly present `yaml` report. I mean in the end, we always go through the list in order anyway.

**Comment by FynnBe** [comment]
Yes, we should implement at least one of these features, I think

**Comment by k-dominik** [comment]
Hey @stuarteberg , do you have any opinion on this? It doesn't change the default behavior, it just adds an output file and gives you the option to start from a certain recipe by giving the `--start-from` argument, which I already found super useful.

**Comment by stuarteberg** [comment]
I love it!

**Pull request merged**